### PR TITLE
Add support for "renderStart" and "noContentAvailable" for adup-tech ads

### PR DESCRIPTION
**Pull request by SteffenAnders**
* Add support for "renderStart" and "noContentAvailable" for adup-tech ads
* Add "https://d.adup-tech.com" (delivery) & "https://m.adup-tech.com" (media) to preconnect hosts

### CODE REVIEW

**Comment by SteffenAnders** [comment]
I signed it!

Company: Ad Up Technology AG
Google Group: ad-up-technology-ag@googlegroups.com
E-Mail: steffen.anders@adup-tech.com

see: https://github.com/ampproject/amphtml/pull/2779

File: ads/aduptech.js [hunk 1]
```diff
@@ -21,14 +21,30 @@ import {loadScript, validateData} from '../3p/3p';
...
+  // load aduptech js api
   loadScript(global, 'https://s.d.adup-tech.com/jsapi', () => {
+
+    // force responsive ads for amp
     data.responsive = true;
-    window.uAd.embed('aduptech', data);
+
+    // ads callback => render start
+    data.onAds = () => {
+      global.context.renderStart();
```
Comment (lannka) [inline ads/aduptech.js]: would this work ? 
```
data.onAds = global.context.renderStart
```

File: ads/aduptech.js [hunk 2]
```diff
@@ -21,14 +21,30 @@ import {loadScript, validateData} from '../3p/3p';
...
+    };
+
+    // no ads callback => noContentAvailable
+    data.onNoAds = () => {
+      global.context.noContentAvailable();
```
Comment (lannka) [inline ads/aduptech.js]: ditto

File: ads/aduptech.js (hunk 1 above)
Comment (SteffenAnders) [inline ads/aduptech.js]: no (see inline doc)

File: ads/aduptech.js (hunk 2 above)
Comment (SteffenAnders) [inline ads/aduptech.js]: yes, fixed

**Comment by SteffenAnders** [comment]
rebased to master

**Comment by SteffenAnders** [comment]
Rebased to master again.
Is there a problem with the CLA? :(

**Comment by SteffenAnders** [comment]
Rebased to master

@jridgewell
it seems that the cla/google status is not passing properly. Did i make any mistakes?

I signed it!

Company: Ad Up Technology AG
Google Group: ad-up-technology-ag@googlegroups.com
E-Mail: steffen.anders@adup-tech.com

**Comment by jridgewell** [comment]
@lannka can manually verify your CLA.

**Comment by lannka** [comment]
@SteffenAnders I couldn't find `Ad Up` in our CLA list. Did you follow the bot's guide to sign the CLA?

**Comment by SteffenAnders** [comment]
I signed it!

**Comment by SteffenAnders** [comment]
Rebased to master

@lannka 
I am a member with my e-mail "address steffen.anders@adup-tech.com" of my coorperations googlegroup "ad-up-technology-ag@googlegroups.com" which signed the CLA. I also used the e-mail address "steffen.anders@adup-tech.com" and my github username "SteffenAnders" as author in the git commit.

PS: My coorperation signed the CLA again, maybe it works now. :)

**Comment by lannka** [comment]
Can you confirm the corp name contains string of  "Ad up"? I don't see any on the list.
+@erwinmombay to back me up

**Comment by SteffenAnders** [comment]
The corp name is "Ad Up Technology AG". What list do you mean? :)

**Comment by SteffenAnders** [comment]
* Rebased to master
* Signed CLA additionaly for myself

**Comment by erwinmombay** [comment]
looks like this is green

**Pull request merged**